artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationSyncFileMessage.java

```diff
@@ -258,7 +258,19 @@ public void decodeRest(final ActiveMQBuffer buffer) {
 
    private void readFile(ByteBuffer buffer) {
       try {
-         fileChannel.read(buffer, offset);
+         if (buffer.hasArray()) {
+            raf.seek(offset);
+            final int position = buffer.position();
+            final int read = raf.read(buffer.array(), buffer.arrayOffset() + position, buffer.remaining());
```

Comment:
This one should use the same optimization on https://github.com/apache/activemq-artemis/pull/2844